finesse/bing_search.py

```diff
@@ -0,0 +1,48 @@
+from azure.cognitiveservices.search.websearch import WebSearchClient
+from msrest.authentication import CognitiveServicesCredentials
+import time
+import statistics
+from joblib import Memory
```

Comment:
FYI, there a convention in Python on how to `style` our import:

[RealPython ](https://realpython.com/absolute-vs-relative-python-imports/#styling-of-import-statements)

Reply:
Thanks for the tip ! I wasn't aware of it.